src/libs/ReportUtils.ts

```diff
@@ -1400,7 +1400,7 @@ function getMostRecentlyVisitedReport(reports: Array<OnyxEntry<Report>>, reportM
         const shouldKeep = !isChatThread(report) || getReportNotificationPreference(report) !== CONST.REPORT.NOTIFICATION_PREFERENCE.HIDDEN;
         return shouldKeep && !!report?.reportID && !!(reportMetadata?.[`${ONYXKEYS.COLLECTION.REPORT_METADATA}${report.reportID}`]?.lastVisitTime ?? report?.lastReadTime);
     });
-    return lodashMaxBy(filteredReports, (a) => new Date(reportMetadata?.[`${ONYXKEYS.COLLECTION.REPORT_METADATA}${a?.reportID}`]?.lastVisitTime ?? a?.lastReadTime ?? '').valueOf());
+    return lodashMaxBy(filteredReports, (a) => [reportMetadata?.[`${ONYXKEYS.COLLECTION.REPORT_METADATA}${a?.reportID}`]?.lastVisitTime ?? '', a?.lastReadTime ?? '']);
```

Comment:
@parasharrajat We need to correct the order here by prioritizing the `lastVisitTime` first because sometimes the report isn't visited but the `lastReadTime` is updated. An example is this issue: when we create the thread, `lastReadTime` is updated to the current time and it's more than `lastReadTime` of the inbox report.

Reply:
The problem here is that removing `lastVisitTime` isn't enough to fix this issue. As I mentioned above, although we don't update `lastVisitTime` of the new thread, `lastReadTime` of this report is still the largest time then the bug still happens when we go back from the search page.